R/ggpairs_add.R

```diff
@@ -116,6 +118,47 @@ add_theme_to_ggmatrix <- function(e1, e2) {
   e1
 }
 
+#' @export
+#' @rdname gg-add
+#' @param rows Integer vector. Rows to modify
+#' @param cols Integer vector. Cols to modify
+#' @param only_fn Character vector. If specified, only subplots
+#'   of that type will be modified (see examples).
+#' @details
+#' \code{add_ggproto_to_ggmatrix} gives you more control to modify
+#'   only some subplots.
+#' @examples
+#' ## modify scale
+#' pm + scale_fill_brewer()
+#' ## only first row
+#' add_ggproto_to_ggmatrix(pm, scale_fill_brewer(), rows = 1)
+#' ## only certain types of subplots
+#' add_ggproto_to_ggmatrix(
+#'   pm, scale_fill_brewer(),
+#'   only_fn = c("ggally_box_no_facet", "ggally_barDiag")
+#' )
+add_ggproto_to_ggmatrix <- function(
+  e1, e2,
+  rows = 1:e1$nrow, cols = 1:e1$ncol,
+  only_fn = NULL) {
```

Comment:
I'm really liking the intent on this parameter.  But the `fn_name` is lost after 1 iteration. 😞 

So this is not possible as the ggplot2 object has been evaluated.
```r
pm <- ggpairs(tips[, 2:3])

# works
pm2 <- add_ggproto_to_ggmatrix(pm, scale_fill_brewer())

# fails
pm3 <- add_ggproto_to_ggmatrix(pm2, scale_color_brewer(), only_fn = "ggally_densityDiag")
```

Reply:
Maybe we can provide a location parameter?

```
add_ggproto_to_ggmatrix <- function(
  e1, e2,
  location = NULL, 
  rows = NULL,
  cols = NULL
)
```

`location` could be any of the following `c("all", "upper", "lower", "diag", "none")`, a matrix of `T`/`F`, (think of any others)?.  If location is supplied, then rows/cols will not be listened to.

Reply:
Very good idea

Reply:
Very good idea

Reply:
Just thinking but the concept of diag or upper us working only with ggpairs results but not with ggduo